services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableQueryManagerImpl.java

```diff
@@ -678,8 +678,16 @@ QuerySpecification addRowLevelFilter(UserInfo user, QuerySpecification query, In
 			} catch (BadSqlGrammarException e) { // table has not been created yet
 				tableBenefactors = Collections.emptySet();
 			}
-			// Get the sub-set of benefactors visible to the user.
+
+			boolean hasNullBenefactors =tableBenefactors.stream().anyMatch(Objects::isNull);

```

Comment:
Per discussion, we think that using a default of -1 would be a cleaner implementation that would avoid awkward database nulls.